lib/tasks/assets.rake

```diff
@@ -0,0 +1,9 @@
+if Rails.env.production?
+  Rake.application.instance_variable_get('@tasks').delete('assets:precompile')
+  namespace :assets do
+    desc "Override asset precompilation task in production"
+    task :precompile do
+      abort "Please use the 'assets' environment for asset precompilation"
+    end
+  end
```

Comment:
Instead of this surgery, can you try to enhance the existing rake task to yell at you and exit 1 in production?

http://ruby-doc.org/stdlib-2.2.4/libdoc/rake/rdoc/Rake/Task.html#method-i-enhance

[Here's an example](https://github.com/ManageIQ/manageiq/blob/d9dd322bc49191f5e8bfd77c78427f5f29cce4e1/lib/tasks/evm_dba.rake#L302)